Core/clim-core/table-formatting.lisp

```diff
@@ -373,101 +390,77 @@ skips intervening non-table output record structures."))
+                           &allow-other-keys)
   (let ((printer (if printer
                      (if presentation-type
-                         #'(lambda (item stream)
-                             (with-output-as-presentation (stream item presentation-type)
-                               (funcall printer item stream)))
+                         (lambda (item stream)
+                           (with-output-as-presentation (stream item presentation-type)
+                             (funcall printer item stream)))
                          printer)
                      (if presentation-type
-                         #'(lambda (item stream)
-                             (present item presentation-type :stream stream))
+                         (lambda (item stream)
+                           (present item presentation-type :stream stream))
                          #'prin1))))
-    (with-keywords-removed (args (:stream :printer :presentation-type
-                                          :cell-align-x :cell-align-y))
-      (apply #'invoke-format-item-list
-             stream
-             #'(lambda (stream)
-                 (map nil
-                      #'(lambda (item)
-                          (formatting-cell (stream :align-x cell-align-x
-                                                   :align-y cell-align-y)
-                            (funcall printer item stream)))
-                      items))
-             args))))
-
-(defmacro formatting-item-list ((&optional (stream t)
-                                           &rest args
-                                           &key x-spacing y-spacing n-columns n-rows
-                                           stream-width stream-height
-                                           max-width max-height
-                                           initial-spacing (row-wise t) (move-cursor t)
-                                           record-type &allow-other-keys)
-                                &body body)
-  (declare (ignore x-spacing y-spacing n-columns n-rows
-                   stream-width stream-height
-                   max-width max-height
-                   initial-spacing row-wise move-cursor
-                   record-type))
-  (setf stream (stream-designator-symbol stream '*standard-output*))
-  (gen-invoke-trampoline 'invoke-format-item-list
-                         (list stream)
-                         args
-                         body))
+    (apply #'invoke-formatting-item-list
+           stream
+           (lambda (stream)
+             (map nil (lambda (item)
+                        (formatting-cell (stream :align-x cell-align-x
+                                                 :align-y cell-align-y)
+                          (funcall printer item stream)))
+                  items))
+           (alexandria:remove-from-plist
+            args :stream :printer :presentation-type
+                 :cell-align-x :cell-align-y))))
```

Comment:
Why? I think `remove-from-plist` is appropriate since the modified plist is not referenced in other places. And even if it were, `let` with `remove-from-plist` (or our own `remove-keywords`) seems more flexible and clearer to me than a specialized binding macro.

Reply:
with-keywords-removed is our internal macro to do exactly that, I don't see a reason why not to use it. That said, if you really have a strong desire to use remove-from-plist anyway, then bind args in a let over apply.

Reply:
>  That said, if you really have a strong desire to use remove-from-plist anyway

Thanks. Long-term, I would like to remove our `remove-keywords` and `with-keywords-removed` and switch to the alexandria versions. Just like we do with `minf`, `lerp`, etc.

> then bind args in a let over apply.

You mean binding the intermediate value instead of computing it within the `apply` form?

Reply:
yes, imo it is easy to miss when put within apply form

Reply:
Changed.